cmd/clusterctl/client/cluster/installer.go

```diff
@@ -90,6 +111,63 @@ func installComponentsAndUpdateInventory(components repository.Components, provi
 	return providerInventory.Create(inventoryObject)
 }
 
+// waitForReady waits till the installed components are ready.
+func (i *providerInstaller) waitForReady(opts InstallOptions) error {
+	// If we dont have to wait for providers to be installed
+	// return early.
+	if !opts.WaitProviders {
+		return nil
+	}
+
+	log := logf.Log
+	log.Info("Waiting for providers to be available...")
+
+	return i.waitManagerDeploymentsReady(opts)
+}
+
+// waitDeploymentsReady waits till the installed deployments are ready.
```

Comment:
```suggestion
// waitManagerDeploymentsReady waits till the installed deployments are ready.
```

Reply:
I would suggest to use waitForProvidersReady, because it provides sense of what we are achieving without surfacing how we are doing this (which might change in the future)

Reply:
I am going to rename the `waitForReady` function to `waitForProvidersReady` function which will then call `waitManagerDeploymentsReady`. This way the sense of what we are trying to achieve is clear and also the implementation has the flexibility to change in the future.

Reply:
ok, fine for me